__tests__/integration/app.test.js

```diff
@@ -0,0 +1,58 @@
+const app = require("../../app");
+const request = require("supertest");
+const fact = require("../../models/fact");
+const api = require("../../routes/api");
+
+describe("Integration/unit testing app.js", function () {
+  describe("GET /", function () {
+    test("GET request responds with a 200 status code and returns HTML", async function (done) {
+      const response = await request(app).get("/");
+      const { statusCode, text } = response;
+      expect(statusCode).toBe(200);
+      const titleDescription = `An API for interesting facts about numbers`;
+      expect(text).toContain(titleDescription);
+      done();
+    });
+
+    test("GET request should call fact.getFact()", async function (done) {
+      const getFact = jest.spyOn(fact, "getFact");
+      const response = await request(app).get("/");
+      expect(getFact).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  describe("POST => /submit", function () {
+    test("POST request should respond with a 200 status code", async function (done) {
+      const response = await request(app).post("/submit");
+      const { statusCode } = response;
+      expect(statusCode).toBe(200);
+      done();
+    });
+
+    test("POST /submit request should not return data to user if no headers are set", async function (done) {
```

Comment:
is this actually what you're testing? it seems like it returns data to the user even without setting any headers.

Reply:
ah, so if I send this in as part of the request body without headers, the route returns an empty object. But if I _do_ send in the request with header `Content-Type: Application/json`, along with json enabled in the app, then this _does_ work. 

I'm not really sure what the best option would be :|

Reply:
hmm, as long as using the json middleware doesn't impact other routes, maybe the easiest thing is to just include it then?